File: torch/test_torch.py
#from __future__ import print_function

import torch

if __name__ == '__main__':
    x = torch.rand(5, 3)
    print(x)

    print(torch.cuda.is_available())

    print(torch.empty(5, 3))

    print(torch.zeros(5, 3, dtype=torch.long))

    t = torch.tensor([1.2, 2.1])
    print(t)

    print(t.size())

    if torch.cuda.is_available():
        device = torch.device("cuda")
        y = torch.ones_like(x, device=device)
        x = x.to(device)
        z = x + y
        print(z)
        print(z.to("cpu", torch.double))

File: torch/test_torch.cc
#include <torch/torch.h>
#include <iostream>

int main() {
  torch::Tensor x;
  //std::cout << x << std::endl;


  std::cout << torch::cuda::is_available() << std::endl;

  //std::cout << torch::kCUDA << std::endl;

  torch::Tensor y;

  torch::Tensor z = torch::rand({1,1}, torch::kCUDA);
  for(int i = 0; i < 100; ++i)
    {
      x = torch::rand({1, 100000}, torch::kCUDA);
      y = torch::rand({100000, 1}, torch::kCUDA);
      z = z + torch::matmul(x, y);
      std::cout << z << std::endl;
    }

}
